redis/tests/test_cluster_async.rs

```diff
@@ -1807,3 +1808,163 @@ fn test_async_cluster_periodic_checks_update_topology_after_failover() {
     })
     .unwrap();
 }
+
+#[test]
+fn test_async_cluster_periodic_checks_use_management_connection() {
+    let cluster = TestClusterContext::new_with_cluster_client_builder(3, 0, |builder| {
+        builder.periodic_topology_checks(Duration::from_millis(100))
+    });
+
+    block_on_all(async move {
+        let mut connection = cluster.async_connection().await;
+        let mut client_list = "".to_string();
+        let max_requests = 1000;
+        let mut i = 0;
+        loop {
+            if i == max_requests {
+                break;
+            } else {
+                client_list = cmd("CLIENT")
+                    .arg("LIST")
+                    .query_async::<_, String>(&mut connection)
+                    .await
+                    .expect("Failed executing CLIENT LIST");
+                let mut client_list_parts = client_list.split('\n');
+                if client_list_parts
+                .any(|line| line.contains(MANAGEMENT_CONN_NAME) && line.contains("cmd=cluster")) 
+                && client_list.matches(MANAGEMENT_CONN_NAME).count() == 1 {
+                    return Ok::<_, RedisError>(());
+                }
+            }
+            i += 1;
+            let _ = sleep(futures_time::time::Duration::from_millis(10)).await;
+        }
+        panic!("Couldn't find a management connection or the connection wasn't used to execute CLUSTER SLOTS {:?}", client_list);
+    })
+    .unwrap();
+}
+
+fn get_conn_id_from_client_list(connection_name: &str, client_list: &str) -> String {
+    client_list
+        .split('\n')
+        .find(|line| line.contains(connection_name))
+        .expect("Failed finding {connection_name:?} in CLIENT LIST")
+        .split(' ')
+        .next()
+        .expect("Failed getting {connection_name:?} connection ID")
+        .split('=')
+        .nth(1)
+        .expect("Failed getting {connection_name:?} connection ID value")
+        .to_string()
+}
+
+#[test]
+fn test_async_cluster_only_management_connection_is_reconnected_after_connection_failure() {
```

Comment:
this is a huge test. please post some comments throughout, explaining the stages and their goals.